core/variable_operations.cpp

```diff
@@ -297,4 +329,11 @@ Variable reverse(Variable var, const Dim dim) {
 /// Return a deep copy of a Variable or of a VariableProxy.
 Variable copy(const VariableConstProxy &var) { return Variable(var); }
 
+Variable masks_merge(const MasksConstProxy &masks, const Dimensions dims) {
+  auto mask_union = makeVariable<bool>(dims);
```

Comment:
Why do we need `dims` as an argument? This seems weird. Just use dimensions obtained as created from the result, default to 0-D?

Reply:
I store the result in a new Variable, if I default it to 0D with `makeVariable<bool>(false)` it can't do the `operator|=` successfully. 

Shouldn't the result mask Variable have the same dims as the input mask Variables? It doesn't seem like `operator|=` resizes it

Reply:
In-place operators never resize. You need to do `result = result | other` to broadcast the dimensions as required.